db/db_impl/db_impl.h

```diff
@@ -1239,9 +1239,14 @@ class DBImpl : public DB {
   static Status TEST_ValidateOptions(const DBOptions& db_options) {
     return ValidateOptions(db_options);
   }
-
 #endif  // NDEBUG
 
+  // In certain configurations, verify that the table/blob file cache only
+  // contains entries for live files, to check for effective leaks of open
+  // files. This can only be called when purging of obsolete files has
+  // "settled," such as during parts of DB Close().
+  void TEST_VerifyNoObsoleteFilesCached(bool db_mutex_already_held) const;
```

Comment:
We might have to move this declaration to the `#ifdef NDEBUG` block above and `#ifdef` the definition+call too to fix those linker errors.

Reply:
Didn't see that db_impl_debug.cc is already in a giant ifdef